23-input-scanning/exercises/04-grep/main.go

```diff
@@ -34,4 +41,29 @@ package main
 // ---------------------------------------------------------
 
 func main() {
+	var (
+		in      = bufio.NewScanner(os.Stdin)
+		lineMap = map[string]bool{}
+		query   = os.Args[1]
+	)
+
+	in.Split(bufio.ScanLines)
+	for in.Scan() {
+		line := in.Text()
+		_, ok := lineMap[line]
+
+		if ok {
+			fmt.Println(line)
```

Comment:
I do not get what this code is doing; I think it that if you lines in an input files are invalid? lets discuss

Reply:
Fixed